### ISSUE TITLE
gemma-4-12B-it: degenerate output on macOS GPU (ML Drift) backend — 12B-specific (E2B & Qwen3-4B are fine)

### ISSUE THREAD

**Issue by john-rocky** [google-ai-edge/LiteRT-LM#2724]
### Summary

Greedy decoding of the published **`litert-community/gemma-4-12B-it-litert-lm`** (`gemma-4-12B-it.litertlm`, int4) produces **degenerate output** on the **macOS GPU (ML Drift / WebGPU) backend** — long runs of a single token (`0000…`, with occasional `1`/`2`), independent of the prompt. Decode **throughput is normal (~48.5 tok/s)**, so the graph runs and streams; only the logits/sampling result is wrong.

Triangulation shows the failure is **specific to the 12B build**: `gemma-4-E2B-it` and `Qwen3-4B` `.litertlm` models run **correctly** on the same backend and machine.

### Environment

- **Machine:** Mac Studio, Apple M4 Max (40-core GPU), 128 GB, macOS 27β
- **Runtime:** LiteRT-LM, GPU backend (ML Drift / WebGPU)
- **Model:** `litert-community/gemma-4-12B-it-litert-lm` → `gemma-4-12B-it.litertlm` (int4)

### Symptom

Both driver paths reproduce it (greedy, temp 0):

```
Engine API,   --backend gpu:  OUTPUT: [2221222122212221]   (EOS after ~16 tokens)
LiteRTChat,   greedy:         OUTPUT: [00000000…0001000…]  (640 tokens)
```

The model is **GPU-only** (`--backend cpu` → `INVALID_ARGUMENT: Model requires one of [gpu] but Main backend is CPU`), so a same-file CPU/GPU A/B isn't possible; we triangulated against other models instead.

### Triangulation — 12B-specific, not a platform issue

Same machine, same LiteRT-LM Mac GPU backend, same prompt (`"Explain memory bandwidth in one sentence."`), greedy:

| Model (.litertlm, Mac GPU) | Output | Decode tok/s |
|---|---|---|
| `litert-community/gemma-4-E2B-it` (same family, smaller) | ✅ coherent — *"Memory bandwidth is the rate at which data can be transferred between the CPU/GPU"* | 118.6 |
| `litert-community/Qwen3-4B` (different family) | ✅ coherent — *"`<think>` Okay, the user wants a one-sentence explanation…"* | 106.0 |
| `litert-community/gemma-4-12B-it` | ❌ **degenerate** (`0000…` / `2221…`) | 48.5 |

⇒ The Mac GPU (ML Drift) path is correct for **gemma-4-E2B** and **Qwen3-4B**, and breaks **only for gemma-4-12B**. So it's **not** a general Mac-GPU issue and **not** a general gemma-4 issue — it's specific to the **gemma-4-12B build** (its conversion, or a kernel that only mis-fires at the 12B's shapes/dims on ML Drift).

### Likely area (for triage)

The 12B differs from E2B in ways that stress GPU kernels at larger dims: 48 layers, hidden 3840, **16 attention heads with dual head_dim (sliding 256 / full 512)**, `attention_k_eq_v` (full-attention layers have 1 KV head, value = raw pre-norm `k_proj`), attn scale = 1.0 with per-head Q/K-RMSNorm, dual RoPE (θ=1e4 full-rotate 256 / θ=1e6 partial-0.25 over 512), final logit softcap 30. A **degenerate-from-first-token, throughput-normal** failure points at a **numeric/kernel issue on the GPU decode path** (e.g. overflow/precision in the full-attention 512-head_dim path or the softcap, or a bad weight/scale in the 12B conversion) rather than orchestration.

### Ask

Please check whether the published `gemma-4-12B-it.litertlm` (a) reproduces on your ML Drift / Android GPU path or is Mac-GPU-only, and (b) whether re-converting fixes it. The model-card benchmarks imply it was validated on device — so a **Mac-GPU-specific numeric regression in a 12B-only kernel** is the leading hypothesis.
